### DISCUSSION TITLE
GitHub Copilot Nightly replaced by pre-release extension

### DISCUSSION THREAD

**Discussion by hsenag** [community/community#59964] [Copilot Conversations]
Soon after the GitHub Copilot technical preview launched (two years ago now!) we started shipping our nightly builds as "GitHub Copilot Nightly" to provide a way for people who like to live on the bleeding edge to try out our latest updates early.

A while after that, the Visual Studio Code team added official support for ["pre-release" extensions](https://code.visualstudio.com/api/working-with-extensions/publishing-extension#prerelease-extensions). However, we had lots of other things to be doing and never got round to moving over.

At long last, thanks to @sandy081 on the Visual Studio Code team who has done the work inside the Copilot codebase, we've made the switch. "GitHub Copilot Nightly" is now deprecated and people who have it installed should soon be migrated over to the new pre-release of "GitHub Copilot" automatically.

This should all be seamless but if you do have any problems please let us know!

**Comment by veverkap** [comment]
![bye-bye-bye-justin-timberlake-gif-gif](https://github.com/community/community/assets/22348/4f36b2fe-cc4f-4ef2-b9d3-4b410c28a342)

**Reply by alirezaraha3** [reply]
[@]()__

**Comment by dhiwakarK** [comment]
Can the customer use the regular VS Code instead of the Nightly with this `pre-release extension` @hsenag ? 🙏🏼 

👀 @trip-t

**Reply by hsenag** [reply]
[Yep](https://code.visualstudio.com/api/working-with-extensions/publishing-extension#prerelease-extensions) 
> Users can choose to install pre-release versions of extensions in *VS Code* or VS Code Insiders

**Reply by dhiwakarK** [reply]
Oh wow, nice! thanks for confirming :)

**Comment by eburnette** [comment]
In VSCode I had this setting to make it always bring in Copilot with chat to all my dev containers:
```
"dev.containers.defaultExtensions": [
  "GitHub.copilot-nightly"
]
```
If I change that to just "GitHub.copilot" it wants to use the production version of the Copilot extension (without chat). How do I make it use the prerelease version automatically?

**Reply by sandy081** [reply]
You can set to `GitHub.copilot@prerelease` for using pre release

**Comment by ericd-mst-github** [comment]
Hi,

When I attempt to switch to the 'pre-release' setting from the CoPilot Extension listing page in VS Code, I run into the following error:

`Can't install pre-release version of 'github.copilot-chat' extension because it is not compatible with the current version of Visual Studio Code (version 1.80.0).`

Any ideas I should try? I have done a few restarts of VS Code already but run into the same issue. I am just trying to enable the CoPilot Chat

**Reply by eburnette** [reply]
I had that and to fix it I had to go back to VS Code Insiders edition.
As of this writing,
- Copilot prerelease needs VS Code Insiders
- Copilot release works with regular VS Code

**Reply by KXwsc** [reply]
The highest version of VS Code Insiders is 1.80, how to install the pre-release version of GitHub Copilot? ?

**Reply by eburnette** [reply]
Currently, I'm using these versions together:

- VSCode 1.81.0-insider 
- GitHub.copilot v1.95.253 (pre-release)
- GitHub.copilot-chat v0.5.2023071105 (pre-release)

I'll be glad when I can just use the production version of everything again because the frequent updates are inconvenient.

**Comment by andrijana44** [comment]
Ich habe keine Ahnung wirklich 🤷‍♀️😒

**Reply by alirezaraha3** [reply]
[__]()
